Utils/src/CandidateTrackFilter.cc

```diff
@@ -297,12 +297,14 @@ bool CandidateTrackFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::
 	//-------------------------------------------------------------------------------------------------
 	loopOverCollection(infos,pfCandidates,ttBuilder,primaryVertex,true);
```

Comment:
You should add an isValid to the lostTracks and pfCandidates as well. Also, you should print out a warning if these collections are missing. Use the edm::LogWarning messageLogger. What do you thin, make sure this doesn't get printed every event, but only the first event where the collection is missing?

Reply:
right now i haven't worried about printing messages every event or not

Reply:
If we don't mind additional variables in the class, I guess printing the error message just for the first event should be fine. Let me see.